dot/sync/worker_pool.go

```diff
@@ -162,8 +162,10 @@ func (s *syncWorkerPool) submitRequest(request *network.BlockRequestMessage,
 
 	if who != nil {
 		syncWorker := s.workers[*who]
-		syncWorker.queue <- task
-		return
+		if syncWorker != nil {
```

Comment:
would there ever be a nil `*syncWorker` in this map?

Reply:
Yeah it appears so. I was panicking because of this and thus added the check

Reply:
I would assume it's because we don't check that the map contains the key of `*who`.  It will default to zero value of `*syncWorker` if the key doesn't exist which would be `nil` in this case.  I think it makes more sense to do a key check instead of checking if `nil`.  A `nil` worker would probably lead me to believe it's a a programmer error, and I would panic.